pkg/hostagent/hostagent.go

```diff
@@ -816,15 +838,26 @@ func forwardSSH(ctx context.Context, sshConfig *ssh.SSHConfig, port int, local,
 }
 
 func (a *HostAgent) watchCloudInitProgress(ctx context.Context) {
+	exitReason := "Cloud-init monitoring completed successfully"
+	var cmd *exec.Cmd
+
+	defer func() {
+		if cmd.Process != nil {
+			logrus.Debug("Cleaning up cloud-init monitoring process")
+			cmd.Process.Kill()
```

Comment:
This code will never be executed because it won't reach here until the `tail` process stops in the VM.

Reply:
confirmed that `ssh` process for `tail` killed. :+1: